lib/Dialect/Dialect.cpp

```diff
@@ -281,7 +281,8 @@ bool llvm_dialects::detail::isOperationDecl(llvm::StringRef fn,
     if (!fn.starts_with(mnemonic))
       return false;
 
-    return fn[mnemonic.size()] == '.';
+    return fn.size() + 2 >= mnemonic.size() && fn[mnemonic.size()] == '_' &&
```

Comment:
Shouldn't this be `fn.size() > mnemonic.size() + 2`? (Strict > makes sense because something has to be after the underscores, though it doesn't *really* matter)

Reply:
Agree that it doesn't matter much. But before something like lgc.foobar. was resulted in returning true, so I'd err on the side of caution here and also let this match for lgc.foobar__ now